cloud/docker/docker_network.py

```diff
@@ -0,0 +1,312 @@
+
+            ipam_config = utils.create_ipam_config(driver=self.parameters.ipam_driver,
+                                                   pool_configs=ipam_pools)
+
+            if not self.check_mode:
+                resp = self.client.create_network(self.parameters.network_name,
+                                                  driver=self.parameters.driver,
+                                                  options=self.parameters.driver_options,
+                                                  ipam=ipam_config)
+
+                self.existing_network = self.client.inspect_network(resp['Id'])
+
+            self.results['actions'].append("Created network %s with driver %s" % (self.parameters.network_name, self.parameters.driver))
+            self.results['changed'] = True
+
+    def remove_network(self):
+        if self.existing_network:
+            self.disconnect_all_containers()
+            if not self.check_mode:
+                self.client.remove_network(self.parameters.network_name)
+            self.results['actions'].append("Removed network %s" % (self.parameters.network_name,))
+            self.results['changed'] = True
+            self.existing_network = None
+
+    def is_container_connected(self, container_name):
+        container_names = [c['Name'] for c in self.existing_network['Containers'].values()]
+        return container_name in container_names
+
+    def connect_containers(self):
+        for name in self.parameters.connected:
+            if not self.is_container_connected(name):
+                if not self.check_mode:
+                    self.client.connect_container_to_network(name, self.parameters.network_name)
+                self.results['actions'].append("Connected container %s" % (name,))
+                self.results['changed'] = True
+
+    def disconnect_missing(self):
+        for c in self.existing_network['Containers'].values():
+            name = c['Name']
+            if name not in self.parameters.connected:
+                self.disconnect_container(name)
+
+    def disconnect_all_containers(self):
+        containers = self.client.inspect_network(self.parameters.network_name)['Containers']
+        for cont in containers.values():
+            self.disconnect_container(cont['Name'])
+
+    def disconnect_container(self, container_name):
+        if not self.check_mode:
+            self.client.disconnect_container_from_network(container_name, self.parameters.network_name)
+        self.results['actions'].append("Disconnected container %s" % (container_name,))
+        self.results['changed'] = True
+
+    def present(self):
+        if self.parameters.force:
+            self.remove_network()
+
+        self.create_network()
+        self.connect_containers()
+        if not self.parameters.incremental:
+            self.disconnect_missing()
+
+        self.results['network'] = self.get_existing_network()
```

Comment:
I think this should be self.results['facts'] rather than self.results['network']. This is the info we want to become part of ansible_facts.